theme-utils.mjs

```diff
@@ -1180,228 +1367,265 @@ Host wpcom-sandbox
 				detached: true,
-			})
+			} );
 		} else {
 			/*
-			* Determines the shell to execute the command.
-			* - Prioritizes using the user's default shell unless it's fish, a known problematic shell.
-			* - In this case, falls back to `/bin/bash` for better syntax compatibility.
-			*/
+			 * Determines the shell to execute the command.
+			 * - Prioritizes using the user's default shell unless it's fish, a known problematic shell.
+			 * - In this case, falls back to `/bin/bash` for better syntax compatibility.
+			 */
 			let shellPath = process.env.SHELL || '/bin/bash';
-			if (shellPath.includes('fish') && fs.existsSync('/bin/bash')) {
+			if (
+				shellPath.includes( 'fish' ) &&
+				fs.existsSync( '/bin/bash' )
+			) {
 				shellPath = '/bin/bash';
 			}
 
-			child = spawn(shellPath, ['-c', command], {
-				stdio: [process.stdin, 'pipe', 'pipe'],
+			child = spawn( shellPath, [ '-c', command ], {
+				stdio: [ process.stdin, 'pipe', 'pipe' ],
 				detached: true,
-			});
+			} );
 		}
 
-		var timer = setTimeout(() => {
+		var timer = setTimeout( () => {
 			try {
-				process.kill(-child.pid, 'SIGKILL');
-			} catch (e) {
-				console.log('Cannot kill process');
+				process.kill( -child.pid, 'SIGKILL' );
+			} catch ( e ) {
+				console.log( 'Cannot kill process' );
 			}
-		}, timeout);
+		}, timeout );
 
-		child.stdout.on('data', (data) => {
+		child.stdout.on( 'data', ( data ) => {
 			response += data;
-			if (logResponse) {
-				console.log(data.toString());
+			if ( logResponse ) {
+				console.log( data.toString() );
 			}
-		});
+		} );
 
-		child.stderr.on('data', (data) => {
+		child.stderr.on( 'data', ( data ) => {
 			errResponse += data;
-			if (logResponse) {
-				console.log(data.toString());
+			if ( logResponse ) {
+				console.log( data.toString() );
 			}
-		});
+		} );
 
-		child.on('exit', (code) => {
-			clearTimeout(timer)
-			if (code !== 0) {
-				reject(errResponse.trim());
+		child.on( 'exit', ( code ) => {
+			clearTimeout( timer );
+			if ( code !== 0 ) {
+				reject( errResponse.trim() );
 			}
-			resolove(response.trim());
-		});
-	});
+			resolove( response.trim() );
+		} );
+	} );
 }
 
-async function escapePatterns() {
-	// get staged files
-	const staged = (await executeCommand(`git diff --cached --name-only`)).split('\n');
-	// get unstaged, untracked files
-	const unstaged = (await executeCommand(`git ls-files -m -o --exclude-standard`)).split('\n');
-	
-	// avoid duplicates and filter pattern files
-	const patterns = [...new Set([...staged, ...unstaged])].filter(file => file.match(/.*\/patterns\/.*.php/g));
-
-	// arrange patterns by theme
-	const themePatterns = patterns.reduce((acc, file) => {
-		const themeSlug = file.split('/').shift();
+async function escapePatterns( directory ) {
+	let patternFiles;
+
+	if ( directory ) {
+		// If a directory is provided, use fast-glob to find all PHP files in the specified directory
+		patternFiles = await glob( `${ directory }/**/*.php`, {
```

Comment:
If we wanted to keep it `npm run patterns:escape -- [argument]` I think it would make more sense to accept the theme folder name / slug as the argument, since that's how the other theme-utils scripts work.

Reply:
>  I think it would make more sense to accept the theme folder name / slug as the argument rather than a specific directory

That was the intent, but I see that the parameter naming was confusing; I've updated it.